### PR TITLE
few small refactoring

### PR DESCRIPTION
**Pull request by Zangetsu38**
- format: run clang format.
- small refactor firmware install dialog
- small refactor for game install dialog.
- button on setting dialog for download last firmware if it is missing.
- fix popup failed installation with correct typo 
- stub savedatadialogstatus

### CODE REVIEW

File: vita3k/gui/src/game_install_dialog.cpp [hunk 1]
```diff
@@ -33,13 +33,13 @@
...
-nfdchar_t *game_path = nullptr;
-static bool draw_file_dialog = true;
-static bool game_install_confirm = false;
-
 void draw_game_install_dialog(GuiState &gui, HostState &host) {
+    static nfdchar_t *game_path = nullptr;
     nfdresult_t result = NFD_CANCEL;
 
+    static bool draw_file_dialog = true;
+    static bool game_install_confirm = false;
```
Comment (1whatleytay) [inline vita3k/gui/src/game_install_dialog.cpp]: NO GOD PLEASE NO

File: vita3k/gui/src/game_install_dialog.cpp [hunk 2]
```diff
@@ -70,14 +70,14 @@ void draw_game_install_dialog(GuiState &gui, HostState &host) {
         ImGui::Separator();
         ImGui::Spacing();
         ImGui::SetCursorPosX(ImGui::GetWindowWidth() / 2 - 65);
-        if (ImGui::Button("Yes", BUTTON_SIZE)) {
+        if (ImGui::Button("Yes", BUTTON_SIZE))
```
Comment (1whatleytay) [inline vita3k/gui/src/game_install_dialog.cpp]: Why did you remove this? It makes more sense with it... Using just indentation to convey stuff can make things confusing. For multiline ifs, use braces please.

Comment (Zangetsu38) [reply vita3k/gui/src/game_install_dialog.cpp]: What ? Have only one value return on if so is not needed i have juste forget remove it after test, look other part on code already have this removed is just normal

File: vita3k/gui/src/game_install_dialog.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/gui/src/game_install_dialog.cpp]: For ? I have set it here for not set this outside like petmac wanted  and is perfect like this 🤔

**Comment by Zangetsu38** [comment]
> I don't really want aes to be refactored. It's already not reflecting our style, and it won't be reflecting our style after we remove random indentation. I would leave it alone.
> 
> The indentation actually makes it look better imo...
> Screen Shot 2019-11-05 at 2 36 41 PM

I have no refactor this, i have just run clang format because pr for pup no have run format on it that all

**Comment by Zangetsu38** [comment]
i have push fix, for failed change typo on openpup for failed installation

File: vita3k/gui/src/settings_dialog.cpp [hunk 3]
```diff
@@ -170,9 +179,13 @@ void draw_settings_dialog(GuiState &gui, HostState &host) {
...
         } else {
-            ImGui::TextColored(GUI_COLOR_TEXT, "No modules present.\nPlease install the firmware");
-            if (ImGui::IsItemHovered())
-                ImGui::SetTooltip("Put decrypted modules inside 'vs0/sys/external/'.");
+            ImGui::TextColored(GUI_COLOR_TEXT, "No modules present.\nPlease download and install the last firmware.");
+            std::ostringstream link;
+            if (ImGui::Button("Download Firmware")) {
+                std::string firmware_url = "https://www.playstation.com/en-us/support/system-updates/ps-vita/";
+                link << OS_PREFIX << firmware_url;
+                system(link.str().c_str());
```
Comment (1whatleytay) [inline vita3k/gui/src/settings_dialog.cpp]: AHHH `system`!

Zang pointed out in dms that this is already used in about_dialog.cpp. I guess it is okay.

I would've probably used `fmt::format("{} https://www.playstation.com/en-us/support/system-updates/ps-vita/", OS_PREFIX)` though.
